evm/src/cpu/kernel/asm/core/precompiles/rip160.asm

```diff
@@ -0,0 +1,37 @@
+global precompile_rip160:
+    %stack (address, retdest, new_ctx, kexit_info, gas, address, value, args_offset, args_size, ret_offset, ret_size) ->
+        (new_ctx, kexit_info, ret_offset, ret_size)
+    DUP1
+    SET_CONTEXT
+    // stack: (empty)
+    PUSH 0x100000000 // = 2^32 (is_kernel = true)
```

Comment:
If we create a `kexit_info` to use with `%charge_gas`, I think we need to initialize `kexit_info` with the gas that the parent context gave us (from metadata)?

If it's easier we could probably skip the `is_kernel` bit, as it should only be needed if we call a syscall. (In fact I think we can probably get rid of the `is_kernel` bit entirely, as AFAIK we don't have any kernel code that calls a syscall, and doing so may be weird since most syscalls charge gas.)

Alternatively, we could not construct `kexit_info`, but directly calculate `leftover_gas = our_gas_from_metadata - gas_to_charge`. Since that wouldn't involve `%charge_gas`, we'd need to check for out-of-gas ourselves, but it might still be a tad simpler than constructing `kexit_info`?

Reply:
> If we create a kexit_info to use with %charge_gas, I think we need to initialize kexit_info with the gas that the parent context gave us (from metadata)?

Hm I'm confused. I thought `kexit_info>>192` holds the gas used so far in this context (which is 0 here) and that the gas limit is stored in the context metadata. Is that not the case?

> If it's easier we could probably skip the is_kernel bit, as it should only be needed if we call a syscall. (In fact I think we can probably get rid of the is_kernel bit entirely, as AFAIK we don't have any kernel code that calls a syscall, and doing so may be weird since most syscalls charge gas.)

Ah I was wondering about this. Maybe I can keep it like this until we get rid of the `is_kernel` bit, just to make it less confusing.

Reply:
Oh you're right, forgot about that. Seems fine as is then.